dspace-api/src/main/java/org/dspace/importer/external/endnote/service/EndnoteImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.endnote.service;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.dspace.importer.external.exception.FileSourceException;
+import org.dspace.importer.external.metadatamapping.MetadataFieldConfig;
+import org.dspace.importer.external.metadatamapping.contributor.MetadataContributor;
+import org.dspace.importer.external.service.components.AbstractPlainMetadataSource;
+import org.dspace.importer.external.service.components.dto.PlainMetadataKeyValueItem;
+import org.dspace.importer.external.service.components.dto.PlainMetadataSourceDto;
+
+/**
+ * Implements a metadata importer for Endnote files
+ *
+ * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
+ */
+public class EndnoteImportMetadataSourceServiceImpl extends AbstractPlainMetadataSource {
+
+    @Override
+    public String getImportSource() {
+        return "EndnoteMetadataSource";
+    }
+
+    @Override
+    protected List<PlainMetadataSourceDto> readData(InputStream fileInpuStream) throws FileSourceException {
+        List<PlainMetadataSourceDto> list = new ArrayList<>();
+        try {
+            int lineForDebug = 3;
+            List<PlainMetadataKeyValueItem> tokenized = tokenize(fileInpuStream);
+            List<PlainMetadataKeyValueItem> tmpList = new ArrayList<>();
+            for (PlainMetadataKeyValueItem item : tokenized) {
+                if (item.getKey() == null || item.getKey().isEmpty()) {
+                    throw new FileSourceException("Null or empty key expected on line "
+                    + lineForDebug + ". Keys cannot be null nor empty");
+                }
+                if ("EF".equals(item.getKey())) {
+                    break;
+                }
+                if ("ER".equals(item.getKey())) {
+                    PlainMetadataSourceDto dto = new PlainMetadataSourceDto();
+                    dto.setMetadata(new ArrayList<>(tmpList));
+                    list.add(dto);
+                    tmpList = new ArrayList<>();
+                } else {
+                    if (item.getValue() == null || item.getValue().isEmpty()) {
+                        throw new FileSourceException("Null or empty value expected on line "
+                        + lineForDebug + ". Value expected");
+                    }
+                    tmpList.add(item);
+                }
+                lineForDebug++;
+            }
+        } catch (Exception e) {
+            throw new FileSourceException("Error reading file");
+        }
+        return list;
+    }
+
+
+    private List<PlainMetadataKeyValueItem> tokenize(InputStream fileInpuStream)
```

Comment:
Can we describe this method in JavaDocs.  I know it's a private method, but it's very important to this class so it'd be good to document it.